info.go

```diff
@@ -0,0 +1,141 @@
+/*****************************************************************************
+*   (c) 2020 Copyright, Real-Time Innovations.  All rights reserved.         *
+*                                                                            *
+* No duplications, whole or partial, manual or electronic, may be made       *
+* without express written permission.  Any such copies, or revisions thereof,*
+* must display this notice unaltered.                                        *
+* This code contains trade secrets of Real-Time Innovations, Inc.            *
+*                                                                            *
+*****************************************************************************/
+
+// Package rti implements functions of RTI Connector for Connext DDS in Go
+package rti
+
+// #cgo windows CFLAGS: -I${SRCDIR}/include -I${SRCDIR}/rticonnextdds-connector/include -DRTI_WIN32 -DNDDS_DLL_VARIABLE
+// #cgo linux,arm CFLAGS: -I${SRCDIR}/include -I${SRCDIR}/rticonnextdds-connector/include -DRTI_UNIX -DRTI_LINUX
+// #cgo windows LDFLAGS: -L${SRCDIR}/rticonnextdds-connector/lib/x64Win64VS2013 -lrtiddsconnector
+// #cgo linux,arm LDFLAGS: -L${SRCDIR}/rticonnextdds-connector/lib/armv6vfphLinux3.xgcc4.7.2 -lrtiddsconnector -ldl -lnsl -lm -lpthread -lrt
+// #include "rticonnextdds-connector.h"
+// #include <stdlib.h>
+import "C"
+import (
+	"encoding/json"
+	"errors"
+	"strconv"
+	"unsafe"
+)
+
+/********
+* Types *
+*********/
+
+// Infos is a sequence of info samples used by an input to read DDS meta data
+type Infos struct {
+	input *Input
+}
+
+/********************
+* Private Functions *
+********************/
+
+func newInfos(input *Input) *Infos {
+	// Error checking for the input is skipped because it was already checked
+	return &Infos{
+		input: input,
+	}
+}
```

Comment:
Let's put private function(s) at the bottom

Reply:
I realized most private functions should stay in `rticonnextdds_connector.go` as they are mainly used in the functions in the file. I moved private functions to the bottom.